sdks/java/core/src/main/java/org/apache/beam/sdk/io/util/package-info.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** Defines utilities that ease working with IO in Apache Beam. */
+package org.apache.beam.sdk.io.util;
```

Comment:
can we move `SerializableAvroCodecFactory` inside of the `io` package and make it package-private? That would probably be easiest.

I've found that anything that appears to be part of the public API ends up having users that depend on it, even if we mark it as for internal use only. So the best solution is to use Java visibility to make sure it's a private utility.